spec/filters/truncate_spec.rb

```diff
@@ -29,8 +29,25 @@
 
     stress_it "should truncate all strings in the hash" do
       expect(event.get("[foo][bar]").bytesize).to be <= length
+      if event.get("[foo][bar]").bytesize > length
+        expect(subject).to receive(:filter_matched).once
```

Comment:
I think that the "invoked" expectation should be set before the "execution" of the subject

Reply:
I don't quite follow what you mean here, could you include an example or suggested change?

Reply:
I'm saying that, in a test, we can't invoke a method before declaration of the expectation it is called.

The code is asserting that `#filter_matched` is invoked on `subject` while the `filter_matched` method is invoked by the `subject.filter` in
```ruby
before { subject.filter(event) }
```

So the execution of the `filter_matched` happens before the 
```ruby
expect(subject).to receive(:filter_matched)
```

Could be summarized as:
```ruby
  class AClass
    def a_method
      puts "AClass#a_method invoked"
    end

    def method
      a_method
    end
  end

  describe "#method" do
    subject { AClass.new }
    before { subject.method }

    it "should call method_b" do
      expect(subject).to receive(:a_method)
    end
  end
```

This snippet, once inserted in spec, fails. To make it green we have to switch `before` to `after`.

So it's not clear to me why in the PR the tests doesn't fail while should expect that it fails.